spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -152,8 +152,13 @@ class DeltaCatalog extends DelegatingCatalogExtension
       .getOrElse(spark.sessionState.catalog.defaultTablePath(id))
     val storage = DataSource.buildStorageFormatFromOptions(writeOptions)
       .copy(locationUri = Option(loc))
-    val tableType =
-      if (location.isDefined) CatalogTableType.EXTERNAL else CatalogTableType.MANAGED
+    val isManagedLocation = Option(allTableProperties.get(TableCatalog.PROP_IS_MANAGED_LOCATION))
```

Comment:
TableCatalog also seems to have `PROP_EXTERNAL` ? Is this also serving the same purpose? What happens when both `PROP_EXTERNAL` and `PROP_IS_MANAGED_LOCATION` are set to true?

      /**
       * A reserved property to specify a table was created with EXTERNAL.
       */
      String PROP_EXTERNAL = "external";

Since we are making DeltaCatalog aware of the `TableCatalog.PROP_IS_MANAGED_LOCATION`, should we also make it aware of `PROP_EXTERNAL` for the sake of completeness in this PR itself?

Reply:
+1

I also think we should add asserts and checks here so that UC-Spark catalog never sets these mutually exclusive properly inconsistently. 

same comments i made in the linked PR - https://github.com/unitycatalog/unitycatalog/pull/449/files#r1750747701

Reply:
Spark will never set `PROP_IS_MANAGED_LOCATION` when calling `createTable` of the root catalog. Spark only   sets it during `loadTable` for `V1Table`, so that commands like DESCRIBE TABLE can know it's a managed table even if it has a location.

Reply:
Also see https://github.com/unitycatalog/unitycatalog/pull/449#discussion_r1751065632

Reply:
It is upto the catalog impl to decide how they want to proceed in such scenario? Can we add an assertion in Delta to fail in such a scenario?

Reply:
Delta is more like a proxy and it will eventually forward the request to HMS/UC or whatever custom catalog. I think it's better to leave it to the underlying catalog to enforce it.

Reply:
Anyway, this is not an issue for now as neither HMS nor UC generate location for external tables.

Reply:
It is fine for now. Thats why i merged the PR. But it always good to think of future proofing things so that there are no surprises in the future when somebody also loads DeltaCatalog with other catalogs.